src/EndpointV2/EndpointV2Middleware.php

```diff
@@ -227,10 +227,21 @@ private function bindContextParams(
     private function applyAuthScheme(
         array $authSchemes,
         CommandInterface $command
-    ) : void
+    ): void
     {
         $authScheme = $this->resolveAuthScheme($authSchemes);
-        $command->setAuthSchemes($authScheme);
+
+        $command['@context']['signature_version'] = $authScheme['version'];
+
+        if (isset($authScheme['name'])) {
+            $command['@context']['signing_service'] = $authScheme['name'];
+        }
+
+        if (isset($authScheme['region'])) {
+            $command['@context']['signing_region'] = $authScheme['region'];
+        } elseif (isset($authScheme['signingRegionSet'])) {
+            $command['@context']['signing_region_set'] = $authScheme['signingRegionSet'];
+        }
```

Comment:
Are there any default values for:
- $command['@context']['signing_region']
- $command['@context']['signing_region_set']

Or one of those is guarantee to be always present in the resolved authScheme?

I ask this because, what would happen if none of those conditions are met?

Reply:
If neither of those conditions are met, we just use the signing region (which is always set) from the client config.  This happens [here](https://github.com/aws/aws-sdk-php/blob/fb67a49453c2b1a51268156b70adeba6c619e743/src/AwsClient.php#L412).  We're not doing anything with the region set right now because its presence indicates a `sigv4a` signature version and we hard-code the `*` value into the CRT signing config. This won't change until we start seeing multiple values in region sets.